packages/application/src/frontend.ts

```diff
@@ -276,6 +276,14 @@ export namespace JupyterFrontEnd {
      */
     readonly currentWidget: Widget | null;
 
+    /**
+     * A signal emitted when the focused widget in the application shell changes.
+     */
+    readonly currentChanged?: ISignal<
```

Comment:
Should this new field have the same `### Notes` docstring as above?

Reply:
The top comment of the PR says:

> Small performance gain by avoiding noisy 'many-changed' command signals

Also is the performance gain of this change to the `JupyterFrontEnd.IShell` worth it? This interface has been quite stable since JupyterLab 1, so we should be careful (even though in this case it's optional and backwards compatible)

Reply:
This signal would be useful in a few other core plugins and in the extensions so I am in favour of adding it.

I agree that it needs better docs, in particular what is the expected behaviour for single widget apps such Notebook.

Reply:
The doc string has been expanded in 528124e

Reply:
FWIW noisy signals can cause issues beyond small performance impacts; for example in https://github.com/jupyterlab/jupyterlab/issues/15465#issuecomment-1989046706 I tracked down a browser crash caused by a rendering bottleneck caused by the use of `ActivateRequest` where `labShell.currentChanged` would suffice (but was not used presumably because it was not on `IShell`).